From: Thomas Schwinge <thomas@schwinge.name>
Subject: [PATCH] t/__i686_defined

Reported roughly a thousand times on glibc mailing lists (with different kinds
of patches).

First: <http://sources.redhat.com/ml/libc-alpha/2002-10/msg00157.html>

<http://sourceware.org/bugzilla/show_bug.cgi?id=411>

---

    ../sysdeps/i386/fpu/s_frexp.S:66: Error: invalid identifier for ".ifdef"
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: junk at end of line, first unrecognized character is `1'
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: junk at end of line, first unrecognized character is `1'
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: junk at end of line, first unrecognized character is `1'
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: junk at end of line, first unrecognized character is `.'
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: junk at end of line, first unrecognized character is `1'
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: expected comma after name `' in .size directive
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: ".endif" without ".if"
    ../sysdeps/i386/fpu/s_frexp.S:66: Error: junk `.get_pc_thunk.dx' after expression
    make[2]: *** [/home/thomas/data-local/tmp/gnu-0/src/glibc.obj/math/s_frexp.os] Error 1
    make[2]: Leaving directory `/media/data-local/thomas/tmp/gnu-0/src/glibc/math'
    make[1]: *** [math/subdir_lib] Error 2
    make[1]: Leaving directory `/media/data-local/thomas/tmp/gnu-0/src/glibc'
    make: *** [all] Error 2

 sysdeps/i386/sysdep.h |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/sysdeps/i386/sysdep.h
+++ b/sysdeps/i386/sysdep.h
@@ -22,6 +22,13 @@
 
 #ifdef	__ASSEMBLER__
 
+/* This macro will be defined to 1 if GCC has been configured with
+   --with-arch=i686, causing build failures with __i686.get_pc_thunk.  */
+#if defined __i686
+#undef __i686
+#define __i686 __i686
+#endif
+
 /* Syntactic details of assembler.  */
 
 #ifdef HAVE_ELF
